### ISSUE TITLE
Postgres unique constraint violation repo_external_service_unique_idx

### ISSUE THREAD

**Issue by chrismwendt** [sourcegraph/sourcegraph#2690]
- Visit https://sourcegraph.com/github.com/lukakerr/Pine
- Notice the error:

![image](https://user-images.githubusercontent.com/1387653/54175467-422b9f80-4448-11e9-914b-66cf9166eab4.png)

Trace: https://app.lightstep.com/sourcegraph-prod/trace?span_guid=608ddd6925155445&at_micros=1552366323485676

```
pq: duplicate key value violates unique constraint "repo_external_service_unique_idx"
```

**Comment by keegancsmith** [comment]
This is because we do not support renames yet. However, we have added some of our constraints in for the new syncer behind a feature flag. So this is probably due to our migration "picking" the wrong repository in its migration as the new one. I took a look at our pgsql logs, and this constraint is firing relatively often. I think this issue is unique to Sourcegraph.com, WDYT @tsenart.

If it is unique to Sourcegraph.com it will be fixed soon as we work on #2025. In fact I planned to work on this today. If it ends up being delayed, we can investigate removing the constraint just for Sourcegraph.com and redoing the migration manually once we are ready.

If it isn't unique to Sourcegraph.com we should look into dropping this constraint before branch cut.
